Protocol.md

```diff
@@ -0,0 +1,177 @@
+
+## Description
+
+The two parties exchange a set of three messages, and terminates when the non-initiating party broadcasts the Fund Transaction to the Bitcoin network.
+The following figure gives a brief overview of the protocol:
+
+    +-------+                    +-------+
+    |       |                    |       |
+    |       |---- (1) offer  --->|       |
+    |       |                    |       |
+    |       |<--- (2) accept ----|       |
+    |   A   |                    |   B   |
+    |       |---- (3) sign   --->|       |
+    |       |                    |       |
+    |       |                    |  (4) broadcast fund-tx
+    |       |                    |       |
+    +-------+                    +-------+
+
+### Offer
+
+The initiating party starts the protocol by sending an `offer` message to the other party.
+This message includes the following information:
+1. Contract information (an array of triples defined below)
+1. Oracle information (optional)
+1. A's fund amount (satoshi)
+1. A's public key
+1. A's inputs
+1. A's change outputs (optional)
+1. Estimatesmartfee (satoshi/kbyte)
+1. CET CSV delay (optional)
+1. Refund locktime
+
+#### Contract information
+Contract information consists of a list of triples to be used to create CETs.
+The triple consists of:
+`message` - `a_output` (satoshi) - `b_output` (satoshi)
+where `message` is the data on which the oracle will be providing a (set of) signatures, and `a_output` and `b_output` represent the amount that A and B will receive for this outcome respectively.
+The interpretation of the message semantics is left to the application.
+
+#### Oracle information
+The oracle R point to use to build the CETs as well as meta information to identify the oracle to be used in the contract:
+`R_point` - `meta data`
+The meta data can either be human readable for a user to be able to interpret it, or machine readable so that an application can interpret it and present it to the user.
+If both parties already have this information, transmission is unnecessary.
+
+#### A's fund amount (satoshi)
+How much A inputs into the contract.
+
+#### A's public key
+The public key to be used in the Fund transaction, refund transaction and CETs.
+
+#### A's inputs
+The set of UTXOs to be used as input to the fund transaction.
+
+#### A's change output (optional)
+If the sum of A's inputs is greater than the fund amount plus the fee for the fund transaction, A can specify a change output to be added to the fund transaction.
+
+#### Estimatesmartfee (satoshi/kbyte)
+The fee rate to be used when computing fees for the different transactions.
+
+#### CET CSV Delay (optional)
+The number to use as input to the CSV in the CET transactions.
+This is optional, and a default of 144 (roughly one day) can be use which should give enough time for a party to broadcast its closing transaction and get it included in the blockchain in most circumstances.
```

Comment:
I still think that it makes more sense just to use the same locktime as refund here since these funds should be locked up until the oracle broadcasts a signature anyway so it isn't really hurting the counter-party if you broadcast early and can't follow through